### PR TITLE
Support for PyTorch model

### LINKED ISSUE #343: Replace custom fork of Cloudpickle with the maintained pip package
We're currently using a custom fork of cloudpickle copied into the Clipper repo. But we don't do anything special with our fork, and the project is actively maintained.

### PR DESCRIPTION
**Pull request by Corey-Zumar**
@haofanwang Added some updates to fix the class serialization issue and accidentally closed #322 in the process. Take a look and let me know if you have any questions. 

This should be good to go once @dcrankshaw takes a final pass.

Fixes #343, #314

### CODE REVIEW

File: clipper_admin/clipper_admin/deployers/cloudpickle.py [hunk 1]
```diff
@@ -1,15 +1,20 @@
 """
```
Comment (Corey-Zumar) [inline clipper_admin/clipper_admin/deployers/cloudpickle.py:92]: Pulled the latest cloudpickle. This fixes issues with recursive dependencies that caused issues when serializing pytorch models.

File: clipper_admin/clipper_admin/deployers/pytorch.py [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+
+    # save Torch model
+    torch_weights_save_loc = os.path.join(serialization_dir,
+                                          PYTORCH_WEIGHTS_RELATIVE_PATH)
+
+    torch_model_save_loc = os.path.join(serialization_dir,
+                                        PYTORCH_MODEL_RELATIVE_PATH)
+
+    try:
+        torch.save(pytorch_model.state_dict(), torch_weights_save_loc)
```
Comment (Corey-Zumar) [inline clipper_admin/clipper_admin/deployers/pytorch.py:175]: We now save the model weights and class definition separately. Weights are saved via PyTorch, and the class is pickled with CloudPickle

File: clipper_admin/clipper_admin/deployers/cloudpickle.py (hunk 1 above)
Comment (dcrankshaw) [reply clipper_admin/clipper_admin/deployers/cloudpickle.py:92]: If we're replacing Cloudpickle, let's just switch to using the cloudpickle pip package (`pip install cloudpickle`).

File: dockerfiles/PyTorchContainerDockerfile [hunk 3]
```diff
@@ -0,0 +1,20 @@
+ARG CODE_VERSION
+FROM clipper/py-rpc:${CODE_VERSION}
+
+COPY clipper_admin/clipper_admin/python_container_conda_deps.txt /lib/
+
+RUN echo deb http://ftp.de.debian.org/debian jessie-backports main >> /etc/apt/sources.list \
+      && apt-get update --fix-missing \
+      && apt-get install -yqq -t jessie-backports openjdk-8-jdk \
+      && conda install -y --file /lib/python_container_conda_deps.txt \
+      && conda install pytorch torchvision -c soumith
```
Comment (dcrankshaw) [inline dockerfiles/PyTorchContainerDockerfile]: They created a PyTorch conda channel recently, so this should be `conda install pytorch torchvision -c pytorch`

**Review by dcrankshaw — CHANGES_REQUESTED** [review]
Two small changes then this is good to go. @Corey-Zumar can you go ahead and make those?

File: clipper_admin/clipper_admin/deployers/cloudpickle.py (hunk 1 above)
Comment (dcrankshaw) [reply clipper_admin/clipper_admin/deployers/cloudpickle.py:92]: See #343

Comment (Corey-Zumar) [reply clipper_admin/clipper_admin/deployers/cloudpickle.py:92]: Done

File: dockerfiles/PyTorchContainerDockerfile (hunk 3 above)
Comment (Corey-Zumar) [reply dockerfiles/PyTorchContainerDockerfile]: Done

**Comment by Corey-Zumar** [comment]
jenkins test this please

**Review by dcrankshaw — APPROVED** [review]
lgtm

**Pull request merged**